core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -47,6 +47,16 @@ public class PartitionsTable extends BaseMetadataTable {
         new Schema(
             Types.NestedField.required(1, "partition", Partitioning.partitionType(table)),
             Types.NestedField.required(4, "spec_id", Types.IntegerType.get()),
+            Types.NestedField.required(
+                9,
+                "last_updated_at",
+                Types.TimestampType.withZone(),
+                "Partition last updated timestamp"),
+            Types.NestedField.required(
+                10,
+                "last_updated_snapshot_id",
```

Comment:
a) Is it a good idea to keep the snapshot id? Because regularly running expire_snapshots can clean up the snapshots and we may not be able to map what operation these files were created from, even with the snapshot id. 


b) There was also an ask for "latest sequence number" associated with that partition from the community users during partition stats discussion. 

Do you think modified time is enough and no need for the sequence number?

Reply:
also cc: @szehon-ho

Reply:
Hm, what do we think to have both?  (and null if its expired, as per discussion here https://github.com/apache/iceberg/pull/7581#discussion_r1201257740).  If we have it, I feel its more useful than the time, but agree we don't always have it.

I think sequence number will be good too, but do we mean fileSequenceNumber or dataSequenceNumber.  Maybe worth another pr if there's more discussion there?

Reply:
I am just worried that most of the snapshots will be expired and we end up not using that field much. 
The main purpose of storing the snapshot id is for finding what operation has last updated this partition id? In that case, we can store the operation type itself directly maybe. 

> I think sequence number will be good too, but do you mean fileSequenceNumber or dataSequenceNumber? Maybe worth another pr if there's more discussion there?

I guess it is fileSequenceNumber.

Yeah, we can have a separate discussion. I think `data_file_size_in_bytes` per partition can also be one more good candidate for storing here.

Reply:
@RussellSpitzer @aokolnychyi any thoughts here, what would make more sense on Partition table?

Reply:
After some thoughts, I think both will be null if the snapshot is expired.  But I think we are no worse off than the alternative, which is to join the entries + snapshots table for the user to find the last update time.

And I think for this case (lastUpdateTime, lastSnapshotId), it is ok to proceed, typically snapshots live several days at least, and I would imagine the user of the tool is interested in the last updated partition.